ddl/column.go

```diff
@@ -1319,6 +1341,13 @@ func (w *worker) doModifyColumn(
 	if err := adjustColumnInfoInModifyColumn(job, tblInfo, newCol, oldCol, pos, ""); err != nil {
 		return ver, errors.Trace(err)
 	}
+	convertedFromAutoInc := updateAutoRandID > 0 && mysql.HasAutoIncrementFlag(oldCol.Flag)
+	if convertedFromAutoInc {
+		err := migrateAutoIDToAutoRandID(d, t, dbInfo, tblInfo)
```

Comment:
how could this can happen that a table has autoRandom allocator and autoincrement flag here?

Reply:
- `auto_random` column must be the primary key.
- If an `auto_increment` is not the primary key, it cannot change to an `auto_random` column(see `checkAutoRandom`).

As a result, it is not possible to convert the `auto_increment` column on an existing `auto_random` table.